int32 piece1
int32 piece2
---
